dbms/src/Parsers/ParserSelectQuery.cpp

```diff
@@ -186,6 +187,8 @@ bool ParserSelectQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
             return false;
     }
 
+    bool limit_with_ties_occured = false;
```

Comment:
Is it redundant in presense of `select_query->limit_with_ties`?

Reply:
https://github.com/ClickHouse/ClickHouse/blob/5a399c78dc934a93529419b4711af979d5b80fc0/dbms/src/Parsers/ParserSelectQuery.cpp#L108

You can see this peace of code above. So, queries like ```SELECT [DISTINCT] [TOP N [WITH TIES]] expr list LIMIT BY``` are possible.

Reply:
I’ll add something like this into test.

Reply:
Ok, I understand. You can write a comment that is needed for TOP.

Reply:
That's my fault. Because of https://github.com/ClickHouse/ClickHouse/blob/5548c3328d7b97aea3456a5fcc5ddc8210b5e9f6/dbms/src/Parsers/ParserSelectQuery.cpp#L193 it is really redudant. And ``` SELECT TOP n``` equals to ```SELECT ... LIMIT n```. Is is strange that we can use LIMIT with LIMIT BY, but can not use TOP with LIMIT.

Reply:
TOP is totally eqivalent to LIMIT. It doesn't make sense to use both.

Reply:
That means this exception works wrong.